Revise the review policy (#33)
r-multiverse/r-multiverse.github.io: review.md

1. Includes an R package at the top level whose package name is the same as the repository name.
1. Is listed in the `URL` field of the corresponding CRAN page (if a package with the same name is on CRAN).

and if the R package:

1. Includes a license from the "Recommended licenses" section at the end of this policy.
1. Does not have an advisory in the [R Consortium Advisory Database](https://github.com/RConsortium/r-advisory-database).
1. Is not part of the CRAN mirror at <https://github.com/cran>.

## Manual review

R-multiverse [moderators](governance.md#moderator) review [pull requests](https://github.com/r-multiverse/contributions/pulls) that the bot flags for manual review.
The [moderator](governance.md#moderator) inspects the package for compliance with R-multiverse policies, including but not limited to [Acceptable Use](aup.md), [Terms of Use](terms.md), and [Code of Conduct](conduct.md).
The moderator accepts the contribution if and only if it complies with all policies.

## Removal

R-multiverse may remove a package from its own repositories at any time if the package violates R-multiverse policies.

## Modification

> No R-multiverse staff member (administrator, moderator, or otherwise) may modify a package in R-multiverse without the explicit consent of the owners declared in the license of the package.

Reviewer: Let's just remove this section altogether. It is so outside the expected behaviour of a repository that I don't think we need to cover this scenario.

Author: But it's not unheard of in the traditional repositories by any means. I think folks who are aware of past instances will want to know R-multiverse will never meddle with package source code.

Reviewer: A vanishingly small number of people likely know about those instances. I think merely mentioning this is on balance harmful as readers might remember this possibility and might remember seeing it on R-multiverse, even if we say this will never happen.

Author: That's fair. And I guess the package licenses and https://github.com/r-multiverse/production/blob/main/LICENSE.note legally prevent us from doing so anyway.

Author: Removed in https://github.com/wlandau/r-multiverse.github.io/commit/f6dc363996b03781075829a820a81b48cdf0f4ce

Reviewer: Thanks!

## Recommended licenses

The [Acceptable Use Policy](aup.md) prohibits packages that "violate any applicable laws, regulations, or third-party rights, including intellectual property rights".

> In practice, this means each package must have a valid open-source license.

Reviewer: ```suggestion
If a package has a valid open-source license, this ensures that distribution by R-multiverse does not violate the intellectual property rights of the package authors.
```

Author: On second thought, I think we need to  require a valid OSS license. Otherwise, we don’t know if R-multiverse has permission to distribute the package. We already have one package with “Licence: TBD”. We will need clear policy language that prohibits this if we are going to enforce proper guardrails.

Author: @shikokuchuo, I am proposing we codify this via https://github.com/wlandau/r-multiverse.github.io/commit/275a247fa0487c42d27cd3bd96e5e12b471c8f6a. When we say "protect the intellectual property rights of the package authors", we also mean "protect R-multiverse staff from the appearance of copyright infringement".

The following is a list of valid open-source licenses that the bot automatically accepts during reviews.

> If the package includes one of the license below, it implies that the authors consent to distribute the package in R-multiverse.

Reviewer: ```suggestion
```

Author: If anyone objects to their package being distributed on R-multiverse, we can point to the license, and/or we can invite them to submit a PR to r-multiverse/contributions to have it removed from community.

> * [Artistic 2.0](https://opensource.org/license/artistic-2-0)

Reviewer: ```suggestion
* [Apache 2.0](https://opensource.org/license/apache-2-0)
* [Artistic 2.0](https://opensource.org/license/artistic-2-0)
```

title: "Review Policy"
---

This policy dictates the process by which R-multiverse reviews and accepts contributed R packages.

> All contributions must comply with R-multiverse policies, including but not limited to [Acceptable Use](https://r-multiverse.org/aup.html), [Terms of Use](https://r-multiverse.org/terms.html), and [Code of Conduct](https://r-multiverse.org/conduct.html).

Reviewer 2: Use relative links rather than absolute links, or Quarto document links like @shikokuchuo does? https://github.com/r-multiverse/r-multiverse.github.io/blob/6e3040bddda216739ee7098fa4caec3c3186ceb2/contributors.md?plain=1#L74

Author: Good idea, changed in https://github.com/wlandau/r-multiverse.github.io/commit/fc8e265b9025eb21eaf5fba629916465b8815ad4.

1. Was [created by the GitHub web interface](https://r-multiverse.org/contributors.html).
1. Adds new contributed listings to the [`packages` folder](https://github.com/r-multiverse/contributions/tree/main/packages).
1. Does not add, modify, or delete any other files in <https://github.com/r-multiverse/contributions>.

and if each new [contributed listing](https://github.com/r-multiverse/contributions/tree/main/packages):

1. Is a single line of text with a valid HTTPS URL.
1. Points to an existing public GitHub/GitLab repository.

and if the contributed GitHub/GitLab repository:

1. Includes an R package at the top level whose package name is the same as the repository name.
1. Includes a license from the "Recommended licenses" section at the end of this policy.
1. Includes a GitHub/GitLab [release](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases).
1. Is not part of the CRAN mirror at <https://github.com/cran>.
1. Is mentioned in the `URL` field its CRAN page (if on CRAN).

and if the author of the [pull request](https://github.com/r-multiverse/contributions/pulls):

> 1. Is a [public member](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership) of one of the trusted [GitHub organizations](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/about-organizations) listed at <https://github.com/r-multiverse/contributions/blob/main/organizations>.

Reviewer 2: the last link is broken

Author: Just added an new org list at https://github.com/r-multiverse/contributions/blob/main/organizations. Many more to come, but it's a start.

In addition, the GitHub/GitLab repository owner must consent to contribute the repository to R-multiverse. Any of the following is sufficient evidence of consent:

1. The author of the [pull request](https://github.com/r-multiverse/contributions/pulls) is the GitHub owner of the repository.

> 1. The author of the [pull request](https://github.com/r-multiverse/contributions/pulls) is a [public member](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership) of the GitHub organization that owns the repository.

Reviewer 2: I'm now tempted to wonder whether that criterium is not strict enough for an organization like rOpenSci with many members and many different projects

Author: Good point, removed in https://github.com/wlandau/r-multiverse.github.io/commit/95dd3238a96f4fe4eb6bf7be8ea5b82599fc8613. I think this is fine because I expect the majority of packages will come from individual users, not orgs. And for orgs, it's not too much to ask to put an R-multiverse URL in the `DESCRIPTION`.

Author: Actually, as @shikokuchuo pointed out in https://github.com/r-multiverse/help/issues/87#issuecomment-2482364546, this question legally comes down to the license. I attempted to make this clear in https://github.com/wlandau/r-multiverse.github.io/commit/6650d559ac6d154f968c778ddf917048f34b7562. And I like how it simplifies the process.
